src/components/SelectOption/SelectOption.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import usersFromServer from '../../api/users';
+
+interface Props {
+  name: string;
+  value?: string;
+  label: string;
+  placeholder: string;
+  hasError?: boolean;
+  onSelectedUser: React.Dispatch<React.SetStateAction<number>>;
+}
+
+export const SelectOption: React.FC<Props> = ({
+  label,
+  placeholder,
+  hasError = false,
+  onSelectedUser,
+}) => {
+  return (
+    <div className="field">
+      <label className="label" htmlFor="title">
+        {label}
+      </label>
+      <select
+        data-cy="userSelect"
+        defaultValue="0"
+        onChange={event => onSelectedUser(Number(event.target.value))}
```

Comment:
[CODE KNOWLEDGE] - When using array indexes as values in a select option, you are creating a dependency on the array order, which can lead to errors if the array changes. It's better to use the user.id as the value, as it is likely to be unique and consistent.